### PR TITLE
docs(discount-codes): Write documentation for discount code generator

### PR DESCRIPTION
**Pull request by wizzy25**
affects: @commercetools/discount-code-generator

### CODE REVIEW

File: docs/cli/discount-code-generator.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+# Discount Code Generator
+
+Generate unique discount codes to be imported to the [commercetools platform](https://dev.commercetools.com/). See usage below
+
+## Usage
+`npm install @commercetools/discount-code-generator --global`
+
+### CLI
+```
+Usage: discountCodeGenerator [options]
```
Comment (hisabimbola) [inline docs/cli/discount-code-generator.md]: I thought you changed this to `discount-code-generator` in the bin file...

File: docs/cli/discount-code-generator.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+Options:
+  --help, -h                 Show help text.                                                     [boolean]
```
Comment (hisabimbola) [inline docs/cli/discount-code-generator.md]: I expect some basics flag like `--debug` or `logLevel` if needed...

File: docs/cli/discount-code-generator.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  --version, -v              Show version number.                                                [boolean]
```
Comment (hisabimbola) [inline docs/cli/discount-code-generator.md]: does `-v` accepts Boolean? what happens if `false` is passed in?

File: docs/cli/discount-code-generator.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  --quantity, -q             Quantity of discount codes to generate. (Between 1 and 500000)     [required]
+  --code-length, -l          Length of the discount codes to generate.                       [default: 11]
+  --code-prefix, -p          Prefix for each code. No prefix will be used if omitted.        [default: ""]
+  --input, -i                Path to code options CSV or JSON file.
+  --output, -o               Path to store generated output file.                      [default: "stdout"]
+  --delimiter, -d            Used CSV delimiter for input and/or output file.               [default: ","]
+  --multivalueDelimiter, -m  Used CSV delimiter in multivalue fields for input/output file. [default: ";"]
+```
+
+#### Info on flags
```
Comment (hisabimbola) [inline docs/cli/discount-code-generator.md]: Awesome explanation 💯

File: docs/cli/discount-code-generator.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+- The ```--quantity``` flag represents the number of codes to be generated. It must be a number between 1 and 500000
+- The ```--code-length``` flag is used to customise the discount codes according to individual use cases. (more info in Exmples)
+- The ```--input``` flag specifies the path to a CSV or JSON file containing the options the discount codes should have. If this flag is omitted, discount codes without any attributes will be generated.
+- The ```--output``` flag specifies where to output the generated codes. Several notes on this flag:
```
Comment (hisabimbola) [inline docs/cli/discount-code-generator.md]: _...where to output/save the generated codes..._

File: docs/cli/discount-code-generator.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+  - If the file specified already exists, it will be overwritten.
+  - The default location for status report logging is the standard output.
+  - If no output path is specified, the generated codes will be logged to the standard output as a result, status reports will be logged to a `discountCodeGenerator.log` file
+- The ```--delimiter``` flag specifies the delimiter used in the input file and/or output file if any or both are CSV. Defaults to ```','``` if omitted and will be ignored if neither input nor output is CSV
```
Comment (hisabimbola) [inline docs/cli/discount-code-generator.md]: any special reason for using three backticks, I think one does the same job...

File: docs/cli/discount-code-generator.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+- The ```--multivalueDelimiter``` flag specifies the delimiter for arrays in CSV. Note that only the ```cartDiscounts``` field if present should be an array. Defaults to ```';'``` if omitted
```
Comment (hisabimbola) [inline docs/cli/discount-code-generator.md]: There is actually no array in the CSV, it's just a delimiter for multi values in a cell... if you get what I mean...

File: docs/cli/discount-code-generator.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+  --input, -i                Path to code options CSV or JSON file.
+  --output, -o               Path to store generated output file.                      [default: "stdout"]
+  --delimiter, -d            Used CSV delimiter for input and/or output file.               [default: ","]
+  --multivalueDelimiter, -m  Used CSV delimiter in multivalue fields for input/output file. [default: ";"]
+```
+
+#### Info on flags
+- The ```--quantity``` flag represents the number of codes to be generated. It must be a number between 1 and 500000
+- The ```--code-length``` flag is used to customise the discount codes according to individual use cases. (more info in Exmples)
+- The ```--input``` flag specifies the path to a CSV or JSON file containing the options the discount codes should have. If this flag is omitted, discount codes without any attributes will be generated.
```
Comment (hisabimbola) [inline docs/cli/discount-code-generator.md]: An array of codes will be outputed right?

File: docs/cli/discount-code-generator.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+  --code-prefix, -p          Prefix for each code. No prefix will be used if omitted.        [default: ""]
+  --input, -i                Path to code options CSV or JSON file.
+  --output, -o               Path to store generated output file.                      [default: "stdout"]
+  --delimiter, -d            Used CSV delimiter for input and/or output file.               [default: ","]
+  --multivalueDelimiter, -m  Used CSV delimiter in multivalue fields for input/output file. [default: ";"]
+```
+
+#### Info on flags
+- The ```--quantity``` flag represents the number of codes to be generated. It must be a number between 1 and 500000
+- The ```--code-length``` flag is used to customise the discount codes according to individual use cases. (more info in Exmples)
```
Comment (hisabimbola) [inline docs/cli/discount-code-generator.md]: ...little typo here... _Exmples_

also you can make _Examples_ an hyperlink to the correct section

File: docs/cli/discount-code-generator.md (hunk 5 above)
Comment (hisabimbola) [inline docs/cli/discount-code-generator.md]: I think you need to specify that the type of data generated will depend on the extension of the file passed in here and only `csv` and `json` are supported...

File: docs/cli/discount-code-generator.md (hunk 1 above)
Comment (hisabimbola) [inline docs/cli/discount-code-generator.md]: It's called `discount-code-gen`...

File: docs/cli/discount-code-generator.md (hunk 3 above)
Comment (wizzy25) [inline docs/cli/discount-code-generator.md]: >does -v accepts Boolean? what happens if false is passed in?

Well it does, and if false is passed in, it ignores the flag. Surprisingly, this is the default behaviour of yargs (even our other cli tools using yargs have this behaviour) 🤔

File: docs/cli/discount-code-generator.md (hunk 8 above)
Comment (wizzy25) [inline docs/cli/discount-code-generator.md]: Not really, an array of objects with `code` as the only key:
```
[
  {"code": "CODE_1"},
  {"code": "CODE_2"}
  ...,
  {"code": "CODE_N"}
]
```

File: docs/cli/discount-code-generator.md (hunk 7 above)
Comment (wizzy25) [inline docs/cli/discount-code-generator.md]: 🏁

File: docs/cli/discount-code-generator.md (hunk 2 above)
Comment (wizzy25) [inline docs/cli/discount-code-generator.md]: There is `--logLevel` already. And is `--debug` really needed?

Comment (junajan) [inline docs/cli/discount-code-generator.md]: Hello @wizzy25 .. can you add that `--logLevel` param also here to the documentation?

**Pull request merged**